core/src/main/java/org/testcontainers/dockerclient/DockerClientProviderStrategy.java

```diff
@@ -114,109 +115,34 @@ public static DockerClientProviderStrategy getFirstValidStrategy(List<DockerClie
 
-        String dockerClientStrategyClassName = TestcontainersConfiguration.getInstance().getDockerClientStrategyClassName();
-        return Stream
-                .concat(
-                        Stream
-                                .of(dockerClientStrategyClassName)
-                                .filter(Objects::nonNull)
-                                .flatMap(it -> {
-                                    try {
-                                        Class<? extends DockerClientProviderStrategy> strategyClass = (Class) Thread.currentThread().getContextClassLoader().loadClass(it);
-                                        return Stream.of(strategyClass.newInstance());
-                                    } catch (ClassNotFoundException e) {
-                                        log.warn("Can't instantiate a strategy from {} (ClassNotFoundException). " +
-                                                "This probably means that cached configuration refers to a client provider " +
-                                                "class that is not available in this version of Testcontainers. Other " +
-                                                "strategies will be tried instead.", it);
-                                        return Stream.empty();
-                                    } catch (InstantiationException | IllegalAccessException e) {
-                                        log.warn("Can't instantiate a strategy from {}", it, e);
-                                        return Stream.empty();
-                                    }
-                                })
-                                // Ignore persisted strategy if it's not persistable anymore
-                                .filter(DockerClientProviderStrategy::isPersistable)
-                                .peek(strategy -> log.info("Loaded {} from ~/.testcontainers.properties, will try it first", strategy.getClass().getName())),
-                        strategies
-                                .stream()
-                                .sorted(Comparator.comparing(DockerClientProviderStrategy::getPriority).reversed())
-                )
-                .filter(new Predicate<DockerClientProviderStrategy>() {
-
-                    final Set<Class<? extends DockerClientProviderStrategy>> classes = new HashSet<>();
-
-                    @Override
-                    public boolean test(DockerClientProviderStrategy dockerClientProviderStrategy) {
-                        return classes.add(dockerClientProviderStrategy.getClass());
-                    }
-                })
+        Stream<? extends DockerClientProviderStrategy> configuredDockerClientStrategy = loadConfiguredStrategy();
+
+        Stream<DockerClientProviderStrategy> orderedProvidedStrategies = strategies
+            .stream()
+            .sorted(Comparator.comparing(DockerClientProviderStrategy::getPriority).reversed());
```

Comment:
nit: do we _really_ need these as variables? 😅 They make it hard to follow what is used where, while if they were arguments of `Stream.concat`, it would be clear that the only stream we process is `orderedProvidedStrategies`

Reply:
Also, we don't _have_ to use streams here, may as well go for a List of strategies, something like:
```java
var allStrategies = new ArrayList<DockerClientProviderStrategy>();
allStrategies.add(new EnvironmentAndSystemPropertyClientProviderStrategy());

loadConfiguredStrategy().ifPresent(allStrategies::add);

strategies
              .stream()
              .sorted(Comparator.comparing(DockerClientProviderStrategy::getPriority).reversed())
              .collect(Collectors.toCollection(() -> allStrategies));
```

Reply:
I disagree, for me it made it much easier to follow like this, that's why I did it.
Introducing those variables gives natural breaking points while reading the code that summarizes what happened so far.

That being said, this is obviously down to personal preference and style as well, so we can proceed either way. It helped me a lot to understand the code and make the changes.

Reply:
@bsideup 
So, should we do this additional refactoring in this PR into using lists instead of Streams? Note that we don't have any explicit tests for this class, so we have to be careful to no accidentally introduce a regression.

Reply:
I'd 👍 using lists as well here as well - it's more easily debuggable when looking at the intermediate states.

Reply:
@kiview the refactoring to lists is simple and unlikely to introduce any regression, while being a great improvement to the readability / debuggability, so I suggest we do it :)

Reply:
Alright, will do this.